designs/administrator/future-openshift/bulk-actions/index.md

```diff
@@ -0,0 +1,146 @@
+![toolbar](img/toolbar.png)
+
+Users can click the checkbox in the bulk selector to select all items in the list (a maximum of 200 items can be selected at a time). If all items are selected, the bulk selector checkbox will be filled with a check and the bulk selector will read ‘200 selected’, informing users that the max number of items in the list are selected. Clicking the bulk selector checkbox again will deselect all items.
+![bulk-selector-states](img/bulk-selector-states.png)
+
+Users can also choose to select items in view by choosing the "select view" option within the bulk selector menu. The bulk selector checkbox will be filled in with a '-' to indicate that some items are selected, and the bulk selector will display the number of items selected (e.g. '8 selected'). Clicking the bulk selector checkbox will deselect all items.
+![select-page](img/select-page.png)
+
+Users can also choose to select a group of items by clicking the associated checkboxes next to them. The bulk selector checkbox will be filled in with a ‘-‘ to indicate that some items are selected, and the bulk selector will display the number of items selected (e.g. ‘2 selected’). Clicking the bulk selector checkbox will deselect all items.
+![add-labels](img/add-labels.png)
+
+## Behavior
+If no items are selected, the bulk actions dropdown will be shown as a basic dropdown, but the actions within will be disabled. A tooltip per menu item will explain to users that they must select resources in order to take that specific action.
+![none-checked](img/none-checked.png)
+
+When items are selected, the bulk actions basic dropdown will turn into a primary dropdown, and actions within it will be enabled. 
+![add-labels](img/add-labels.png)
+
+### Item limits
+Depending on the use case, there may be scenarios where users are limited to selecting a certain number of items, assuming there is a large total in a dataset. As a general rule of thumb, the max number of items that can be selected at once will be set at 200. 
+![max-number-200](img/max-number-200.png)
+
+Once the user hits that limit, they will not be able to select more items and all other checkboxes will be disabled. 
+![max-number-error](img/max-number-error.png)
+
+### Permission limits
+Depending on the use case, there may be scenarios where users are limited to a certain set of actions to perform on items. If the user does not have permission to perform a specific action on **any** of the selected items, the action will still be enabled in the dropdown. Once they select the action, the error modal will explain that the action cannot be taken.
+![no-permissions](img/no-permissions.png)
+
+If the user has permission to perform a specific action on **some** of the selected items, the action will be carried out for all of the items that it can be. A failure alert will be included within the modal to list the items that failed to complete.
+![some-permissions](img/some-permissions.png)
```

Comment:
what would happen if there were a lot of items?

Reply:
The modal would scroll / the inline alert would grow with the list of the failures